File: nodes/if-else/if-else.process.py
async def process(inputs, settings, config, nodeConfig):
  # Convert various truthy/falsy values to boolean
  condition = False
  
  if isinstance(inputs.get("condition"), bool):
    condition = inputs.get("condition")
  elif isinstance(inputs.get("condition"), (int, float)):
    condition = inputs.get("condition") != 0
  elif isinstance(inputs.get("condition"), str):
    lowered = inputs.get("condition", "").lower().strip()
    condition = lowered != "" and \
                lowered != "false" and \
                lowered != "0" and \
                lowered != "no" and \
                lowered != "null" and \
                lowered != "undefined"
  else:
    condition = bool(inputs.get("condition"))
  
  # Initialize the output dictionary
  output = {}
  
  # Add result to output only if the corresponding input exists
  if condition:
    if "if_true" in inputs:
      output["result"] = inputs.get("if_true")
    output["true_path"] = True
  else:
    if "if_false" in inputs:
      output["result"] = inputs.get("if_false")
    output["false_path"] = True
  
  return output 

File: nodes/if-else/if-else.process.js
export default async ({inputs, settings, config}) => {

  // Convert various truthy/falsy values to boolean
  let condition = false;
  
  if (typeof inputs.condition === 'boolean') {
    condition = inputs.condition;
  } else if (typeof inputs.condition === 'number') {
    condition = inputs.condition !== 0;
  } else if (typeof inputs.condition === 'string') {
    const lowered = inputs.condition.toLowerCase().trim();
    condition = lowered !== '' && 
                lowered !== 'false' && 
                lowered !== '0' && 
                lowered !== 'no' &&
                lowered !== 'null' &&
                lowered !== 'undefined';
  } else {
    condition = !!inputs.condition;
  }
  
  // Prepare result based on condition
  const result = condition ? inputs.if_true : inputs.if_false;
  
  // Return different outputs based on condition
  if (condition) {
    return {
      result,
      true_path: true,
      // false_path is intentionally not set
    };
  } else {
    return {
      result,
      false_path: true
      // true_path is intentionally not set
    };
  }
}; 